eras/shelley/impl/src/Cardano/Ledger/Shelley/LedgerState/DPState.hs

```diff
@@ -121,24 +149,26 @@ instance NoThunks (InstantaneousRewards c) => NoThunks (DState c)
 instance NFData (InstantaneousRewards c) => NFData (DState c)
 
 instance (CC.Crypto c, ToCBOR (InstantaneousRewards c)) => ToCBOR (DState c) where
-  toCBOR (DState unified fgs gs ir) =
-    encodeListLen 4
+  toCBOR (DState unified fgs gs ir ds) =
+    encodeListLen 5
       <> toCBOR unified
       <> toCBOR fgs
       <> toCBOR gs
       <> toCBOR ir
+      <> toCBOR ds
 
 instance (CC.Crypto c, FromSharedCBOR (InstantaneousRewards c)) => FromSharedCBOR (DState c) where
   type
     Share (DState c) =
       (Interns (Credential 'Staking c), Interns (KeyHash 'StakePool c))
   fromSharedPlusCBOR =
-    decodeRecordNamedT "DState" (const 4) $ do
+    decodeRecordNamedT "DState" (const 5) $ do
       unified <- fromSharedPlusCBOR
       fgs <- lift fromCBOR
       gs <- lift fromCBOR
       ir <- fromSharedPlusLensCBOR _1
-      pure $ DState unified fgs gs ir
+      ds <- fromSharedPlusLensCBOR (_1 . toMemptyLens _1 id)
```

Comment:
This is a complicated way of telling how the deserialisation can share things, so that there is only one copy in memory.
It is complicated because DPState needs to share two different things (Credential 'Staking c) and (KeyHash 'PoolParam c)
this line tells how to separate the two things, so as to pass just  (Credential 'Staking c) sharing to the DState part of DPState.